01-data-structures/06-trees/binary_tree/binary_search_tree.rb

```diff
@@ -1,21 +1,100 @@
 require_relative 'node'
 
 class BinarySearchTree
-
+  attr_reader :root
   def initialize(root)
+    @root = root
   end
 
   def insert(root, node)
+    current_node = root
+    while current_node
+      if (current_node.rating >= node.rating && current_node.left.nil?)
+        current_node.left = node
+        return
+      elsif (current_node.rating < node.rating && current_node.right.nil?)
+        current_node.right = node
+        return
+      elsif (current_node.left && current_node.rating >= node.rating)
```

Comment:
How did you spot this? I know part of it is just experience, but whoa, feels like you've got some kind of jedi powers. Took me a good while of staring at it, taking a break, moving on to something else, to actually see it.

I wouldn't have flagged it. How did you see/spot it.